src/controller/auth.php

```diff
@@ -1175,3 +1183,262 @@
     //redirect
     $this->package('global')->redirect('/auth/login');
 });
+
+/**
+ * Process the SSO Login
+ *
+ * @param Request $request
+ * @param Response $response
+ */
+$cradle->get('/auth/sso/login/:service_type', function ($request, $response) {
+    $type = $request->getStage('service_type');
+
+    // get config
+    $config = $this->package('global')->config('services', $type. '-main');
```

Comment:
Create a `$global` variable to use fewer function calls like `$this->package('global')`